### DISCUSSION TITLE
Is it possible to list API docs grouped by package (or assembly, or project)?

### DISCUSSION THREAD

**Discussion by craigfowler** [dotnet/docfx#8018] [Q&A]
When looking at API docs, the default behaviour for the table of contents is to list the various types, grouped by their namespace.  For larger projects, such as frameworks which include more than one NuGet package, it would be helpful is "The package in which this type appears" were also available as a form of grouping.

For example, I'm writing a library which has an "Abstractions" package (no dependencies, just contains the API definition for consumption) and also a "Main" package (only needs to be referenced in startup projects, to add the framework to DI).  I'd like to be able to show/organise the API docs showing which types are in the Abstractions package and which are in the Main package.

_The perfect solution_ for my use-case organises the types by NuGet package.  That might not be feasible though - and I'm OK with that.  I'm conscious that DocFX would have to be able to understand all of the complexities of NuGet packaging to be 100% sure that it matched the right type with the right package.  Acceptable alternatives which would be _just as good_ for most scenarios are:

* Group by .NET Project
* Group by Assembly Name

I have noticed that the Assembly Name is included in the generated API docs for a type, so that seems plausible.  Does such a feature already exist (and I failed to find it)?  Or should I turn this into a feature-request ticket?

**Comment by frederikprijck** [comment]
This looks a lot like the issue I am having. I opened another discussion on here as I believed it could be different, but I did want to connect the two together.

Would what you are trying come down to what I have described in https://github.com/dotnet/docfx/discussions/8204 ? If so, did you find any way to implement it in such away that it avoids the issue described?

**Reply by craigfowler** [reply]
@frederikprijck - I agree that it looks related but I'm not sure it's the same thing.  It seems your discussion is about something _not working as expected_.  What I'm referring to is a more convenient way to organise/group documentation for types in the table of contents UI.

Currently what I get in a ToC looks a little like:

```text
My.First.Namespace
  TypeA
  TypeB
My.Second.Namespace
  TypeC
```

What I would like is more like the following (assuming that those types were in those assemblies/projects/packages):

```text
AssemblyOne
  My.First.Namespace
    TypeB
AssemblyTwo
  My.First.Namespace
    TypeA
  My.Second.Namespace
    TypeC
```

**Reply by frederikprijck** [reply]
Thanks @craigfowler , that makes sense. My attention was pulled here because of grouping by project vs assemblies, as it looks like I have an issue with files using the same namespace, but inside different assemblies/projects. 

So I had a feeling that same different kind of grouping could be related as once we can group things in the UI like this, I assume #8204 might also work (well, assuming the UI relies on some internal grouping by assembly, which would then, potentially, allow to resolve #8204).
